packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -937,6 +947,14 @@ function isLogicalNegationExpression(
   return node.type === AST_NODE_TYPES.UnaryExpression && node.operator === '!';
 }
 
+function isLogicalExpressionNotNullishCoalescing(
+  node: TSESTree.Node,
+): node is TSESTree.LogicalExpression {
+  return (
+    node.type === AST_NODE_TYPES.LogicalExpression && node.operator !== '??'
```

Comment:
> `&& node.operator !== '??'`

I tried removing this added check locally and no unit tests failed. Either the check is unnecessary or there's a bag in unit test coverage. My hunch is the latter, but haven't looked too deeply. Requesting changes either way.